presto-main/src/main/java/com/facebook/presto/operator/HashAggregationOperator.java

```diff
@@ -360,20 +568,159 @@ protected Page computeNext()
                             BlockBuilder output = pageBuilder.getBlockBuilder(types.size() + i);
                             aggregator.evaluate(groupId, output);
                         }
-
-                        groupId++;
                     }
 
                     return pageBuilder.build();
                 }
             };
         }
+
+        private void rebuildAggregators()
```

Comment:
I don't like `merge`, `not merge` differentiation here.
Maybe it should be differentiated with `partial`, `not partial` aggregations instead (e.g: `rebuildAggregators`, `rebuildPartialAggregators`)?

Anyway, `rebuildAggregatorsForMerge` is a bit cryptic. I see that it defines `numberOfColumnsInKey` parameter. Maybe change the name to be more meaningful and relates to `numberOfColumnsInKey`?